### ISSUE TITLE
"--show-only" shows an error when the template is empty

### ISSUE THREAD

**Issue by dani8art** [helm/helm#7295]
I get an error when using `helm template --show-only templates/test.yaml my-chart` and the manifest generated from this is empty, e.g:

1. Having a template in `templates/secret.yaml` and `values.yaml`:
```yaml
{{- if .Values.secret.enabled }}
apiVersion: v1
kind: Secret
metadata:
  name: my-secret
  labels: 
    app.kubernetes.io/name: my-app
    helm.sh/chart: my-chart-1
type: Opaque
data:
  bind-password: {{ (include "postgresql-ha.ldapPassword" .) | b64enc | quote }}
{{- end -}}
```
```
secret:
  enabled: false
```

2. Executing `helm template --show-only templates/secret.yaml my-chart` I get:
```
Error: could not find template templates/secret.yaml in chart
```

3. And executing `helm template --show-only templates/secret.yaml --set secret.enabled=true my-chart` I get:

```
---
# Source: postgresql-ha/templates/ldap-secrets.yaml
apiVersion: v1
kind: Secret
metadata:
  name: my-secret
  labels: 
    app.kubernetes.io/name: my-app
    helm.sh/chart: my-chart-1
type: Opaque
data:
  bind-password: "YWRtaW4="
```

**should it be an error? Actually, the template is there but it is empty, right?**
**How can I avoid this error message?**

Output of `helm version`:
```bash
version.BuildInfo{Version:"v3.0.1", GitCommit:"7c22ef9ce89e0ebeb7125ba2ebf7d421f3e82ffa", GitTreeState:"clean", GoVersion:"go1.13.4"}
```
Output of `kubectl version`: **not relevant**

Cloud Provider/Platform (AKS, GKE, Minikube etc.): **not relevant**

**Comment by lkysow** [comment]
Worth noting this is different behaviour from Helm 2.

**Comment by bacongobbler** [comment]
> Worth noting this is different behaviour from Helm 2.

This seems intentional. The reason being here is that `helm template -x` would only render a particular template and show the output. `-show-only` renders the full chart, then shows the output of said template. The big difference here being that the old syntax required the file to be present in the chart before it was rendered, and the new flag requires the template to exist after the rendering process. There were cases with the old flag where if a particular template relied on another template to render correctly, the user effectively broke the chart author’s assumptions about how the chart is rendered (like with a library chart as a dependency).

In other words, the behaviour here seems correct from a technical standpoint and it is intended to return an error when a template is empty. What‘s unclear to me is what use case the old behaviour is preventing.

**Comment by lkysow** [comment]
Thanks @bacongobbler for the detailed explanation that makes sense. For context we were using the previous behaviour in our tests:

```bash
@test "client/ClusterRole: enabled by default" {
  cd `chart_dir`
  local actual=$(helm template \
      -x templates/client-clusterrole.yaml  \
      . | tee /dev/stderr |
      yq 'length > 0' | tee /dev/stderr)
  [ "${actual}" = "true" ]
}

@test "client/ClusterRole: disabled with global.enabled=false" {
  cd `chart_dir`
  local actual=$(helm template \
      -x templates/client-clusterrole.yaml  \
      --set 'global.enabled=false' \
      . | tee /dev/stderr |
      yq 'length > 0' | tee /dev/stderr)
  [ "${actual}" = "false" ]
}
```

The second test fails with helm 3.

We'll need to change to expecting the "Could not find template" error.

I took a look through the code and the behaviour of stripping out empty templates is pretty deep so I don't think this would be easy to fix, even if that was desired.

**Comment by bacongobbler** [comment]
closing as stale.

**Comment by Rycieos** [comment]
I found this as a workaround to this bug:

Add these lines to the top of the template in question:
```yaml
---
# This comment is to prevent empty template from erroring on targeted template.
```
Now `helm template --show-only ...` will happily print those lines with no error. This is a valid YAML document, though passing it as input to `kubectl apply` will error, since the document is empty.

**Comment by pshem** [comment]
> I found this as a workaround to this bug:
> 
> Add these lines to the top of the template in question:
> 
> ```yaml
> ---
> # This comment is to prevent empty template from erroring on targeted template.
> ```
> 
> Now `helm template --show-only ...` will happily print those lines with no error. This is a valid YAML document, though passing it as input to `kubectl apply` will error, since the document is empty.

The default behaviour is extremely user hostile. Adding a yaml comment helps - thank you!
